app/js/components/MarginSelectorForm.jsx

```diff
@@ -0,0 +1,124 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {FormGroup, Label, InputGroup, InputGroupAddon, Alert} from 'reactstrap';
+import Input from 'react-validation/build/input';
+import '../../../node_modules/react-datetime/css/react-datetime.css';
+import '../../css/Form.scss';
+import {withNamespaces} from 'react-i18next';
+import {withRouter} from 'react-router-dom';
+import Form from 'react-validation/build/form';
+import {isInteger, required} from '../validators';
+
+class MarginSelectorForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: '',
+      isAbove: true,
+      marketError: '',
+      rate: 0
+    };
+  }
+
+  onRadioButtonChange = (e) => {
+    this.setState({isAbove: e.target.value === "above"});
+  }
+
+  onInputChange = (e) => {
+    this.setState({rate: e.target.value});
+  }
+
+  componentDidMount() {
+    // Redirecting to previous step
+    if(!this.props.fiat) this.props.history.push('/seller/fiat');
+
+    this.setState({
+      isAbove: this.props.margin.isAbove, 
+      rate: this.props.margin.rate
+    });
+  }
+
+  _submit(address){
+    if(!this.validate()) return;    
+    this.props.onSubmit(parseInt(this.state.rate, 10), this.state.isAbove); 
+    this.props.history.push(address);
+  }
+
+  back = () => {
+    this._submit("/seller/fiat");
```

Comment:
Should you validate and submit on back?

Reply:
It can be removed. I added it to store the values on the form if you click back, but now that you mention it, it does not look like normal UX. I'll remove this